docs/null-and-undefined-handling.md

```diff
@@ -0,0 +1,101 @@
+# Handling Null and Undefined Values in Find Operations
+
+TypeORM provides fine-grained control over how `null` and `undefined` values are handled in find operations through two connection-level configuration options: `treatJsNullAsSqlNull` and `throwOnUndefinedInFind`.
+
+## Default Behavior
+
+By default, TypeORM skips both `null` and `undefined` values in where conditions. This means that if you include a property with a `null` or `undefined` value in your where clause, it will be ignored:
+
+```typescript
+// Both queries will return all posts, ignoring the text property
+const posts1 = await repository.find({
+    where: {
+        text: null
+    }
+});
+
+const posts2 = await repository.find({
+    where: {
+        text: undefined
+    }
+});
+```
+
+## Treating JavaScript null as SQL NULL
+
+You can change how `null` values are handled using the `treatJsNullAsSqlNull` option in your connection configuration:
+
+```typescript
+const dataSource = new DataSource({
+    // ... other options
+    treatJsNullAsSqlNull: true
+});
+```
+
+When enabled, JavaScript `null` values in where conditions will be transformed into SQL `NULL` values. For example:
+
+```typescript
+// With treatJsNullAsSqlNull: true
+// This will only return posts where the text column is NULL in the database
+const posts = await repository.find({
+    where: {
+        text: null
+    }
+});
+```
+
+## Throwing on Undefined Values
+
+The `throwOnUndefinedInFind` option allows you to catch potential programming errors by throwing when `undefined` values are encountered in find operations:
+
+```typescript
+const dataSource = new DataSource({
+    // ... other options
+    throwOnUndefinedInFind: true
+});
+```
+
+When this option is enabled, the following error will be thrown if an undefined value is encountered:
+```
+TypeORMError: Undefined value encountered in property 'text' of the find operation. Set 'throwOnUndefinedInFind' to false in connection options to skip properties with undefined values.
+```
+
+Note that this only applies to explicitly set `undefined` values. Properties that are not provided in the where clause are still ignored:
+
+```typescript
+// This will throw an error with throwOnUndefinedInFind: true
+const posts1 = await repository.find({
+    where: {
+        text: undefined
+    }
+});
+
+// This will NOT throw an error, even with throwOnUndefinedInFind enabled
+const posts2 = await repository.find({
+    where: {
+        title: "My Post" // text property is not provided, so it's ignored
+    }
+});
+```
```

Comment:
This explanation of the difference between undefined and missing properties is a little verbose. Is it necessary? 

I should hope it's obvious that where conditions are always partials and there's no need to list every property of the entity, even with `throwOnUndefinedInFind: true`.

Reply:
Agreed, will remove 🙏